beacon/endpoints/rest/response/response_schema.py

```diff
@@ -238,59 +175,35 @@ def build_response(data, qparams, func):
             'beaconHandover': None, # build_beacon_handover
         }
 
-    error = build_error(qparams)
-    if error is not None:
-        response['error'] = error
+    if non_accessible_datasets:
+        response['error'] = build_error(non_accessible_datasets)
 
     return response
 
 
 def build_variant_response(data, qparams):
     """"Fills the `results` part with the format for variant data"""
 
-    variant_requested_schemas = qparams.requestedSchemasVariant[0]
-    variant_annotation_requested_schemas = qparams.requestedSchemasVariantAnnotation[0]
-    # LOG.debug('variant_requested_schemas= %s', variant_requested_schemas)
-    # LOG.debug('variant_annotation_requested_schemas= %s', variant_annotation_requested_schemas)
+    variant_func = qparams.requestedSchema[1]
+    variant_annotation_func = qparams.requestedAnnotationSchema[1]
 
-    variants = get_formatted_content(data, 'Variant', (variant_requested_schemas or []))
-    variant_annotations = get_formatted_content(data, 'VariantAnnotation', (variant_annotation_requested_schemas or []))
-
-    for (v, v_a, row) in itertools.zip_longest(variants, variant_annotations, data):
-        # LOG.debug('v= %s', v)
+    for row in data:
         yield {
-            'variant': v,
-            'variantAnnotations': v_a,
+            'variant': variant_func(row),
+            'variantAnnotations': variant_annotation_func(row),
             'variantHandover': None,  # build_variant_handover
             'datasetAlleleResponses': jsonb(row['dataset_response'])
         }
 
 
-def build_individual_response(data, qparams):
-    """"Fills the `results` part with the format for individual data"""
-
-    individual_requested_schemas = qparams.requestedSchemasIndividual[0]
-    # LOG.debug('individual_requested_schemas= %s', individual_requested_schemas)
-
-    return get_formatted_content(data, 'Individual', (individual_requested_schemas or []))
-
+def build_biosample_or_individual_response(data, qparams):
+    """"Fills the `results` part with the format for biosample or individual data"""
 
-def build_biosample_response(data, qparams):
-    """"Fills the `results` part with the format for sample data"""
+    return build_formatted_response(data, qparams.requestedSchema[1])
 
-    biosample_requested_schemas = qparams.requestedSchemasBiosample[0]
-    # LOG.debug('biosample_requested_schemas= %s', biosample_requested_schemas)
 
-    return get_formatted_content(data, 'Biosample', (biosample_requested_schemas or []))
-
-
-def get_formatted_content(data, field_name, schemas):
-    """Formats the data according to the first requested schema"""
-    # LOG.debug('schemas: %s', schemas)
+def build_formatted_response(data, func):
```

Comment:
No need that function, remove it.

Reply:
Nope, it _is_ used by `build_biosample_or_individual_response`

Reply:
Move the list comprehension where the function is called.
Don't add a useless function call, just for one line.

Your function is saying "I'll make the list comprehension for you, if you pass me the parameters", as opposed to "you make the list comprehension with the parameters at the place where you call the function".

Reply:
That means you might have to update the `build_biosample_or_individual_response` call too.